src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidAccountManagerBroker.cs

```diff
@@ -0,0 +1,452 @@
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenInteractiveParameters acquireTokenInteractiveParameters)
+        {
+
+            CheckPowerOptimizationStatus();
+
+            AndroidBrokerStaticHelper.InteractiveBrokerTokenResponse = null;
+
+            BrokerRequest brokerRequest = BrokerRequest.FromInteractiveParameters(
+                authenticationRequestParameters, acquireTokenInteractiveParameters);
+
+            // There can only be 1 broker request at a time so keep track of the correlation id
+            AndroidBrokerStaticHelper.InteractiveRequestCorrelationId = brokerRequest.CorrelationId;
+
+            try
+            {
+                await InitiateBrokerHandshakeAsync().ConfigureAwait(false);
+                await AcquireTokenInteractiveViaBrokerAsync(brokerRequest).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                _logger.ErrorPiiWithPrefix(ex, "Android broker interactive invocation failed. ");
+                _brokerHelper.HandleBrokerOperationError(ex);
+            }
+
+            using (_logger.LogBlockDuration("Waiting for Android broker response. "))
+            {
+                await AndroidBrokerStaticHelper.ReadyForResponse.WaitAsync().ConfigureAwait(false);
+                return AndroidBrokerStaticHelper.InteractiveBrokerTokenResponse;
+            }
+        }
+
+        public async Task<MsalTokenResponse> AcquireTokenSilentAsync(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenSilentParameters acquireTokenSilentParameters)
+        {
+            CheckPowerOptimizationStatus();
+
+            BrokerRequest brokerRequest = BrokerRequest.FromSilentParameters(
+                authenticationRequestParameters, acquireTokenSilentParameters);
+
+            try
+            {
+                await InitiateBrokerHandshakeAsync().ConfigureAwait(false);
+                return await AcquireTokenSilentViaBrokerAsync(brokerRequest).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                _logger.ErrorPiiWithPrefix(ex, "Android broker silent invocation failed. ");
+                _brokerHelper.HandleBrokerOperationError(ex);
+                throw;
+            }
+        }
+
+        private async Task AcquireTokenInteractiveViaBrokerAsync(BrokerRequest brokerRequest)
+        {
+            using (_logger.LogMethodDuration())
+            {
+                // onActivityResult will receive the response for this activity.
+                // Launching this activity will switch to the broker app.
+
+                _logger.Verbose("Starting Android Broker interactive authentication. ");
+                Intent brokerIntent = await GetIntentForInteractiveBrokerRequestAsync(brokerRequest, _parentActivity).ConfigureAwait(false);
+
+                if (brokerIntent != null)
```

Comment:
As far as I can tell, GetIntentForInteractiveBrokerRequestAsync will either return an intent object or it will throw. So a null check is not needed here?